src/main/java/school/hei/haapi/authentication/AwsCognitoJwtIdToken.java

```diff
@@ -0,0 +1,77 @@
+package school.hei.haapi.authentication;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.proc.BadJOSEException;
+import com.nimbusds.jose.proc.SecurityContext;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.ConfigurableJWTProcessor;
+import java.io.Serializable;
+import java.text.ParseException;
+import java.util.Collection;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.stereotype.Component;
+import school.hei.haapi.exception.NotAuthorizedException;
+
+@Slf4j
+@Component
+public class AwsCognitoJwtIdToken implements Serializable {
+
+  private final JwtConfiguration jwtConfiguration;
+  private final ConfigurableJWTProcessor<SecurityContext> configurableJwtProcessor;
+
+  public AwsCognitoJwtIdToken(
+      final JwtConfiguration jwtConfiguration,
+      final ConfigurableJWTProcessor<SecurityContext> configurableJwtProcessor) {
+    this.jwtConfiguration = jwtConfiguration;
+    this.configurableJwtProcessor = configurableJwtProcessor;
+  }
+
+  public Authentication authenticate(HttpServletRequest request) {
+    String idToken = request.getHeader(this.jwtConfiguration.getHttpHeader());
+    log.info("Token ID : " + idToken);
```

Comment:
NEVER log token information! If anyone access to our log, they could do bad things